content/kubermatic/main/architecture/feature-stages/_index.en.md

```diff
@@ -0,0 +1,39 @@
++++
+title = "Feature Stages"
+date = 2024-10-11T12:11:35+02:00
+weight = 4
+
++++
+
+## Alpha
+
+- Targeted users: internal, or specific customer
+- Disabled by default by its feature gate
+- Enabling the feature may expose bugs
+- No guaranteed support for the feature
+- The feature may change in incompatible ways in a later software release without notice
+- The whole feature can be revoked immediately and without notice
+- Recommended only for testing and providing feedback
+
+
+## Beta / Technical Preview
+
+- Targeted users: experienced KKP administrators
+- Can be enabled or disabled by default by its feature gate
+- Enabling the feature is considered safe
+- Support is guaranteed for the overall feature
+- The schema and/or semantics of objects may change in incompatible ways in later software release
+  - When this happens, we will provide instructions for migrating to the next version, which can require manual work and/or downtime
+- The whole feature can still be revoked, but with prior notice and respecting a deprecation cycle
+- Recommended for only non-business-critical uses, testing usability, performance, and compatibility in real-world environments
+
+
+## General Availability (GA)
+
+- Users: All users
+- The feature is always enabled; you cannot disable it by a feature gate
```

Comment:
KKP does have some feature gates, like HeadlessInstallation / UserClusterMLA / DevelopmentEnvironment / VerticalPodAutoscaler, which are GA and not enabled by default. This document doesn't yet consider optional features which are meant to be optional.

Reply:
Do all the other GA traits apply to these Optional Features? If yes, then I can just add a sub-item here noting that Optional Features are an exception of this. CC @xmudrii

Reply:
Ideally speaking, this shouldn't happen, there should be other ways to handle those features. But that's ideally, and making this work would require API changes. Given that, let's add a sub-item noting about optional features.

Reply:
Added the line